### ISSUE TITLE
Builder/openstack - support to username and password authentication instead of keypair auth

### ISSUE THREAD

**Issue by rguruju** [mitchellh/packer#3930]
Using packer 0.10.1, building a redhat image on openstack. Its successful with keypair. But, we are looking for username password support. Is it a planned feature in future releases?
Tried using ssh_username and ssh_password, but getting the following error : 
```
openstack: Error waiting for SSH: ssh: handshake failed: ssh: unable to authenticate, attempted methods [none publickey], no supported methods remain
==> openstack: Terminating the source server:
```

**Comment by rickard-von-essen** [comment]
@rguruju I just tossed together #3976, it would be great if you could help out verifying that it works for you.

**Comment by rickard-von-essen** [comment]
Fixed in # #3976
